### PR TITLE
Lite Docs and Example Improvements

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/starter/lightning_lite.rst [hunk 1]
```diff
@@ -3,15 +3,14 @@ LightningLite - Stepping Stone to Lightning
 ###########################################
 
 
+:class:`~pytorch_lightning.lite.LightningLite` enables pure PyTorch users to scale their existing code
+on any kind of device while retaining full control over their own loops and optimization logic.
+
 .. image:: https://pl-public-data.s3.amazonaws.com/docs/static/images/lite/lightning_lite.gif
-    :alt: Animation showing how to convert a standard training loop to a Lightning loop
-    :width: 600px
+    :alt: Animation showing how to your PyTorch code to convert to LightningLite.
```
Comment (rohitgr7) [inline docs/source/starter/lightning_lite.rst:10]:
```suggestion
    :alt: Animation showing how to convert your PyTorch code to LightningLite.
```

**Review by rohitgr7 — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -24,6 +24,9 @@ __pycache__/
 *.py[cod]
 *$py.class
 timit_data/
+grid_generated*
+grid_ori*
```
Comment (awaelchli) [inline .gitignore:28]: Let's add it to the gitignore file in the examples, because this is generated by the example, correct?

File: docs/source/starter/lightning_lite.rst [hunk 3]
```diff
@@ -3,15 +3,14 @@ LightningLite - Stepping Stone to Lightning
...
 
 
+:class:`~pytorch_lightning.lite.LightningLite` enables pure PyTorch users to scale their existing code
+on any kind of device while retaining full control over their own loops and optimization logic.
+
 .. image:: https://pl-public-data.s3.amazonaws.com/docs/static/images/lite/lightning_lite.gif
-    :alt: Animation showing how to convert a standard training loop to a Lightning loop
-    :width: 600px
+    :alt: Animation showing how to convert your PyTorch code to LightningLite.
+    :width: 400
```
Comment (awaelchli) [inline docs/source/starter/lightning_lite.rst:11]: this looks a bit too small, the font is smaller than the text. how about 500?

File: pl_examples/basic_examples/README.md [hunk 4]
```diff
@@ -61,11 +61,57 @@ This script shows you how to extract the data related components into a `Lightni
...
 # Distributed Data Parallel (DDP)
-python image_classifier_5_lightning_datamodule.py --trainer.gpus 2 --trainer.strategy 'ddp'
+python mnist_examples/image_classifier_5_lightning_datamodule.py --trainer.gpus 2 --trainer.strategy 'ddp'
+```
+
+______________________________________________________________________
+
+#### AutoEncoder
+
+This script shows you to implement an CNN auto-encoder.
```
Comment (awaelchli) [inline pl_examples/basic_examples/README.md:77]:
```suggestion
This script shows you how to implement a CNN auto-encoder.
```

File: pl_examples/basic_examples/README.md [hunk 5]
```diff
@@ -61,11 +61,57 @@ This script shows you how to extract the data related components into a `Lightni
...
+
+# Distributed Data Parallel (DDP)
+python autoencoder.py --trainer.gpus 2 --trainer.strategy 'ddp'
+```
+
+______________________________________________________________________
+
+#### Backbone Image Classifier
+
+This script shows you to implement a `LightningModule` as a `System`.
```
Comment (awaelchli) [inline pl_examples/basic_examples/README.md:94]:
```suggestion
This script shows you how to implement a `LightningModule` as a `System`.
```
Is "System" meant to be formatted?

File: pl_examples/basic_examples/README.md [hunk 6]
```diff
@@ -61,11 +61,57 @@ This script shows you how to extract the data related components into a `Lightni
...
+A System describes a `LightningModule` which takes a single a `torch.nn.Module` which makes
```
Comment (awaelchli) [inline pl_examples/basic_examples/README.md:95]:
```suggestion
A system describes a `LightningModule` which takes a single `torch.nn.Module` which makes
```
I don't think we have formalized "System" somewhere. It's not a name, right?

File: docs/source/starter/lightning_lite.rst (hunk 1 above)
Comment (tchaton) [reply docs/source/starter/lightning_lite.rst]: Done !

File: pl_examples/basic_examples/README.md [hunk 7]
```diff
@@ -61,11 +61,57 @@ This script shows you how to extract the data related components into a `Lightni
...
+exporting to producion simpler.
```
Comment (awaelchli) [inline pl_examples/basic_examples/README.md:96]:
```suggestion
makes exporting to producion simpler.
```

File: .gitignore (hunk 2 above)
Comment (tchaton) [reply .gitignore:28]: Yes, but if you run this from the root, they are generated at the top. I think it is fine.

File: pl_examples/basic_examples/README.md [hunk 8]
```diff
@@ -61,11 +61,57 @@ This script shows you how to extract the data related components into a `Lightni
...
+
+# Distributed Data Parallel (DDP)
+python backbone_image_classifier.py --trainer.gpus 2 --trainer.strategy 'ddp'
+```
+
+______________________________________________________________________
+
+#### PyTorch Profiler
+
+This script shows you to activate the [PyTorch Profiler](https://github.com/pytorch/kineto) with Lightning.
```
Comment (awaelchli) [inline pl_examples/basic_examples/README.md:113]:
```suggestion
This script shows you how to activate the [PyTorch Profiler](https://github.com/pytorch/kineto) with Lightning.
```

File: docs/source/starter/lightning_lite.rst (hunk 3 above)
Comment (tchaton) [reply docs/source/starter/lightning_lite.rst:11]: @awaelchli Any idea why it doesn't get centered ?

File: pl_examples/basic_examples/README.md (hunk 4 above)
Comment (tchaton) [reply pl_examples/basic_examples/README.md:77]: Done !

File: pl_examples/basic_examples/README.md (hunk 5 above)
Comment (tchaton) [reply pl_examples/basic_examples/README.md:94]: Yes, I would say it is a special term.

File: pl_examples/basic_examples/README.md (hunk 6 above)
Comment (tchaton) [reply pl_examples/basic_examples/README.md:95]: Removed. But I think we should actually formalized the term. The community has been using it for a while now.

**Review by justusschock — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: .azure-pipelines/gpu-tests.yml [hunk 9]
```diff
@@ -108,8 +108,6 @@ jobs:
         bash pl_examples/run_examples.sh --trainer.gpus=1
         bash pl_examples/run_examples.sh --trainer.gpus=2 --trainer.strategy=ddp
         bash pl_examples/run_examples.sh --trainer.gpus=2 --trainer.strategy=ddp --trainer.precision=16
-        bash pl_examples/run_examples.sh --trainer.gpus=2 --trainer.strategy=dp
-        bash pl_examples/run_examples.sh --trainer.gpus=2 --trainer.strategy=dp --trainer.precision=16
```
Comment (awaelchli) [inline .azure-pipelines/gpu-tests.yml:112]: removed, as this PR adds the torchmetrics to some of the examples and this is incompatible with DP. 

Considered options:
- switch to ddp_spawn: not working, because at least one example is returning predictions: 
`predictions = cli.trainer.predict(ckpt_path="best", datamodule=cli.datamodule)`
and predictions will be empty here, causing an error.
- hardcoding dp/ddp_spawn into the examples: not really working, because it would error when trying to run the them on CPU :(

File: docs/source/starter/lightning_lite.rst (hunk 3 above)
Comment (awaelchli) [reply docs/source/starter/lightning_lite.rst]: I couldn't find a way. It's possible that a style in the template prevents us from using any center options. Personally I don't find it that important to center it. If we can center it somehow, it would probably be better to make the background fully white :)

**Review by Borda — APPROVED** [review]
